Sources/xcproj/PBXProjEncoder.swift

```diff
@@ -18,6 +18,10 @@ class PBXProjEncoder {
     var multiline: Bool = true
     
     func encode(proj: PBXProj) -> String {
+        // populate reference cache
+        proj.referenceCache.removeAll()
```

Comment:
I like the idea of having a cache to prevent having to look up for objects every time we need to generate a comment. However, I'm not 100% convinced of the implementation. I think the caching is something we need to speed up writing, thus it shouldn't be part of the project. I think there should be an instance that we instantiate at the moment we start encoding the project, sort of an `ObjectsProvider` that we pass to other objects instead of the `PBXProj` to generate the comments:

```swift
protocol ObjectsProviding {
  func get(reference: String) -> PBXObject
}
class ObjectsProvider: ObjectsProviding {
  let proj: PBXProj
  func get(reference: String) -> PBXObject {}
}
```
And then at then, we could pass that provider to the objects:

```swift
let element = serializable.plistKeyAndValue(provider: ObjectsProviding)
```

Eventually, when we move from arrays to string, we could extend `PBXProj` to conform the `ObjectsProviding` protocol, and pass the project instead. What do you think @yonaskolb?

Reply:
Yeah, sounds a fine approach. I just didn't want to over-engineer this as it's designed as mostly a temporary hotfix

Reply:
Good point, then feel free to leave it like that. What do you think about opening an GitHub issue to tackle this once #136 has been merged? Great job @yonaskolb